header.js

```diff
@@ -86,11 +104,17 @@ var BlockHeader = module.exports = function (data) {
  * @return {BN}
  */
 BlockHeader.prototype.canonicalDifficulty = function (parentBlock) {
+  if ((this._common.hardfork() && !this._common.hardforkGteHardfork(null, 'byzantium')) ||
+    (!this._common.hardfork() && !this._common.hardforkIsActiveOnBlock('byzantium', utils.bufferToInt(this.number)))) {
+    throw new Error('Difficulty validation only supported on blocks >= byzantium')
+  }
+  const hardfork = this._common.hardfork() ? this._common.hardfork() : this._common.activeHardfork(this.number)
+
```

Comment:
For better readability, maybe this can be re-written as follows:

```
const hardfork = this._common.hardfork() || this._common.activeHardfork(this.number)

if (!this._common.hardforkGteHardfork(hardfork, 'byzantium')) {
  throw new Error('Difficulty validation only supported on blocks >= byzantium')
}
```

Reply:
Ah, yes, definitely more elegant and readable, updated.